kythe/java/com/google/devtools/kythe/analyzers/java/AutoValuePlugin.java

```diff
@@ -0,0 +1,296 @@
+    Names names = Names.instance(context);
+    if (Iterables.isEmpty(symtab.getClassesForName(names.fromString(AUTO_VALUE_ANNOTATION)))) {
+      // No AutoValue class detected in compilation; early exit.
+      return null;
+    }
+    javacTrees = JavacTrees.instance(context);
+    types = Types.instance(context);
+    return super.visitTopLevel(compilation, v);
+  }
+
+  @Override
+  public Void visitClassDef(JCClassDecl classDef, Void v) {
+    return handleAutoValue(classDef) ? null : super.visitClassDef(classDef, v);
+  }
+
+  private boolean handleAutoValue(JCClassDecl classDef) {
+    // Check if class was generated by the AutoValueProcessor
+    if (!classDef.getModifiers().getAnnotations().stream()
+        .filter(
+            // Class decorated by @Generated(value = "...AutoValueProcessor")
+            ann ->
+                GENERATED_ANNOTATIONS.contains(annotationName(ann))
+                    && AUTO_VALUE_PROCESSOR.equals(annotationLiteralValue(ann, "value")))
+        .findFirst()
+        .isPresent()) {
+      return false;
+    }
+
+    // Find @AutoValue superclass definition.
+    JCClassDecl autoValueClass = null;
+    ClassType classType = (ClassType) classDef.type;
+    while (autoValueClass == null && classType.supertype_field instanceof ClassType) {
+      classType = (ClassType) classType.supertype_field;
+      JCTree superTypeTree = javacTrees.getTree(classType.asElement());
+      if (superTypeTree instanceof JCClassDecl) {
+        JCClassDecl cls = (JCClassDecl) superTypeTree;
+        if (getAnnotation(AUTO_VALUE_ANNOTATION, cls.getModifiers().getAnnotations()).isPresent()) {
+          autoValueClass = cls;
+        }
+      }
+    }
+    if (autoValueClass == null) {
+      return false;
+    }
+
+    // @AutoValue class generates the implementation class.
+    Optional<KytheNode> autoNode = kytheGraph.getNode(autoValueClass.sym);
+    Optional<KytheNode> genNode = kytheGraph.getNode(classDef.sym);
+    if (autoNode.isPresent() && genNode.isPresent()) {
+      entrySets.emitEdge(autoNode.get().getVName(), EdgeKind.GENERATES, genNode.get().getVName());
+    }
+
+    Map<String, JCMethodDecl> propMethods = new HashMap<>(); // property_name -> getter method
+    Map<String, VName> getterNodes = new HashMap<>(); // property_name -> Kythe getter VName
+    Map<String, VName> properties = new HashMap<>(); // property_name -> Kythe property node VName
+
+    // Find potential propMethods in @AutoValue class.
+    for (JCTree def : autoValueClass.getMembers()) {
+      if (def instanceof JCMethodDecl) {
+        JCMethodDecl method = (JCMethodDecl) def;
+        if (method.getModifiers().getFlags().contains(Modifier.ABSTRACT)
+            && method.getParameters().isEmpty()
+            && !isVoid(method.getReturnType())) {
+          Optional<KytheNode> kytheNode = kytheGraph.getNode(method.sym);
+          if (kytheNode.isPresent()) {
+            String name = method.sym.getSimpleName().toString();
```

Comment:
Do we need to strip off a leading "get" and handle canonical capitalization in that case?

Reply:
nm, it's handled below.